Patcher/Program.cs

```diff
@@ -204,28 +210,37 @@ internal static void Main(string[] args)
                     if (backupFileInfo.Exists)
                         Console.WriteLine($"Backup '{backupFileInfo.Name}' created.");
 
-                    Console.WriteLine("Searching for theme offset..");
+                    Console.WriteLine("Searching for theme offset...");
 
                     var buffer = ms.ToArray();
+
                     var lightOffsets = FindPattern(_lightPattern, buffer).ToList();
                     var darkOffsets = FindPattern(_darkPattern, buffer).ToList();
                     lightOffsets.AddRange(darkOffsets);
 
                     var found = lightOffsets.Any();
-
                     if (!found)
                     {
-                        Console.WriteLine("Could not find the theme offset in the specified executable.");
+                        Console.WriteLine("Error: Could not find the theme offset in the specified executable!");
                         return;
                     }
-
-                    var themeByte = themeName == "dark" ? (byte)0x75 : (byte)0x74;
+                    var foundMoreThanExpected = lightOffsets.Count > 1;
+                    if (foundMoreThanExpected)
+                    {
+                        Console.WriteLine("Warning: Found more than expected theme offsets. Outcome might be unexpected!");
+                        return;
+                    }
+                    var themeBytes = themeName == "dark" ? _darkPattern : _lightPattern;
 
                     foreach (var offset in lightOffsets)
                     {
-                        Console.WriteLine($"Patching theme to {themeName}..");
-                        fs.Position = offset;
-                        fs.WriteByte(themeByte);
+                        Console.WriteLine($"Patching theme to {themeName}...");
+                        var offsetInThemeBytes = offset;
+                        foreach (var themeByte in themeBytes) {
```

Comment:
We don't necessarily need to write all of the bytes in the pattern iirc - the pattern is simply for matching, the only "relevant" byte in the pattern is the first one.

If we were to patch all the bytes in the pattern, we'd have a far bigger surface area that is susceptible to changes in the Unity client, since if any of those bytes change, we could potentially break the application.